File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        pos = 0
        for j in range(1, len(nums)):
            if nums[pos] != nums[j]:
                pos += 1
                nums[pos] = nums[j]
        return pos + 1
    
    
        l = r = 0
        while r < len(nums):
            while (r < len(nums)) and nums[r] == nums[l]:
                r += 1
            l += 1
            if r >= len(nums):
                break
            nums[l] = nums[r]
        
        return l
            
        
        stack=set()
        res=[]
        for n in nums:
            if n in stack:
                continue
            else:
                stack.add(n)
                res.append(n)
        l=len(res)
        i=0
        for i in range(l):
            nums[i]=res[i]
        return l

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        // Edge case: if array is empty, return 0
        if (nums.length == 0) {
            return 0;
        }

        // Initialize the `pre` index to 0
        int pre = 0;

        // Iterate over the array starting from the second element
        for (int i = 1; i < nums.length; i++) {
            // If current element is different from the element at `pre` index
            if (nums[i] != nums[pre]) {
                pre++;
                // Move the unique element to the position right after the last unique element
                nums[pre] = nums[i];
            }
        }
        
        // The length of the array with unique elements
        return pre + 1;
    }
}
